test_harness.py

```diff
@@ -364,27 +365,20 @@ async def test_project_8(dut):
     await wishbone_write(dut, ADDR_PROJECT, project_number)
     assert dut.active_project == project_number
 
-    dut.io_reset = 0
-    dut.io_data = 1
-    dut.io_newData = 1
+    dut.project_io_in[23:8] = 1
+    dut.project_io_in[24] = 1
 
     await ClockCycles(dut.wb_clk_i, 1)
 
-    dut.wbs_dat_i[24] = 0
+    dut.project_io_in[24] = 0
     
     await ClockCycles(dut.wb_clk_i, 16)
 
-    assert dut.wbs_dat_o[16] == True
+    assert dut.project_io_out[16] == True
 
     await ClockCycles(dut.wb_clk_i, 496)
 
-    assert dut.wbs_dat_o[17] == True
+    assert dut.project_io_out[17] == True
 
     await ClockCycles(dut.wb_clk_i, 1000) 
-    
-
-    await ClockCycles(dut.wb_clk_i, 10)
-
-    
-
```

Comment:
Looking good :+1: 

See here on how to do simulated wishbone writes in the testbench:
https://github.com/mattvenn/mpw-multi-project-harness/blob/f347770296d75ce813705a6bc8b2ebe49ae85681/test_harness.py#L142

Reply:
So... hm... now I am starting to realize (maybe). If I use wishbone, that's where I get the clock from. So, if I am choosing to go without wishbone, I should be getting clock from outside, as well?

How would I go about that in cocotb?

Reply:
Correct, I'd connect the clock and reset pins to `project_io_in` too. Because you'll need it to clock in data into `io_data`.

maybe like this:

```verilog
    .clock(project_io_out[7][35]),
    .reset(reset | project_io_out[7][36]),
    .io_data(project_io_in[7][23:8]),
    .io_newData(project_io_in[7][24]),
    .io_red(project_io_out[7][32:25]),
    .io_hSync(project_io_out[7][33]),
    .io_vSync(project_io_out[7][34])
```

all IO pads used :D

Reply:
Ok, I think I got it right this time.